posthog/hogql/database/schema/util/session_where_clause_extractor.py

```diff
@@ -0,0 +1,390 @@
+            elif node.op == CompareOperationOp.Lt or node.op == CompareOperationOp.LtEq:
+                return ast.CompareOperation(
+                    op=ast.CompareOperationOp.LtEq,
+                    left=ast.ArithmeticOperation(
+                        op=ast.ArithmeticOperationOp.Sub,
+                        left=rewrite_timestamp_field(node.left, self.context),
+                        right=ast.Call(name="toIntervalDay", args=[ast.Constant(value=SESSION_BUFFER_DAYS)]),
+                    ),
+                    right=node.right,
+                )
+        elif is_right_timestamp_field and is_left_constant:
+            # let's not duplicate the logic above, instead just flip and it and recurse
+            if node.op in [
+                CompareOperationOp.Eq,
+                CompareOperationOp.Lt,
+                CompareOperationOp.LtEq,
+                CompareOperationOp.Gt,
+                CompareOperationOp.GtEq,
+            ]:
+                return self.visit(
+                    ast.CompareOperation(
+                        op=CompareOperationOp.Eq
+                        if node.op == CompareOperationOp.Eq
+                        else CompareOperationOp.Lt
+                        if node.op == CompareOperationOp.Gt
+                        else CompareOperationOp.LtEq
+                        if node.op == CompareOperationOp.GtEq
+                        else CompareOperationOp.Gt
+                        if node.op == CompareOperationOp.Lt
+                        else CompareOperationOp.GtEq,
+                        left=node.right,
+                        right=node.left,
+                    )
+                )
+
+        return ast.Constant(value=True)
+
+    def visit_arithmetic_operation(self, node: ast.ArithmeticOperation) -> ast.Expr:
+        # don't even try to handle complex logic
+        return ast.Constant(value=True)
+
+    def visit_not(self, node: ast.Not) -> ast.Expr:
+        return ast.Constant(value=True)
+
+    def visit_call(self, node: ast.Call) -> ast.Expr:
+        if node.name == "and":
+            return self.visit_and(ast.And(exprs=node.args))
+        elif node.name == "or":
+            return self.visit_or(ast.Or(exprs=node.args))
+        return ast.Constant(value=True)
+
+    def visit_field(self, node: ast.Field) -> ast.Expr:
+        return ast.Constant(value=True)
+
+    def visit_constant(self, node: ast.Constant) -> ast.Expr:
+        return ast.Constant(value=True)
+
+    def visit_placeholder(self, node: ast.Placeholder) -> ast.Expr:
+        raise Exception()  # this should never happen, as placeholders should be resolved before this runs
+
+    def visit_and(self, node: ast.And) -> ast.Expr:
+        exprs = [self.visit(expr) for expr in node.exprs]
+
+        flattened = []
+        for expr in exprs:
+            if isinstance(expr, ast.And):
+                flattened.extend(expr.exprs)
+            else:
+                flattened.append(expr)
```

Comment:
It could be possible to have even deeper nested `and(and(and(...), ..), ..)` calls. Making this flatten deeper could save a support hero some time in the future.

Reply:
I believe it should flatten an infinitely nested AND, I just added a test for this